integration/solana/simple_collectible.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// DISCLAIMER: This file is an example of how to mint and transfer NFTs on Solana. It is not production ready and has not been audited for security.
+// Use it at your own risk.
+
+import '../../solana-library/spl_token.sol';
+
+contract SimpleCollectible {
+    // On Solana, the mintAccount represents the type of token created. It saves how many tokens exist in circulation.
+    address private mintAccount;
+    // The public key for the authority that should sign every change to the NFT's URI
+    address private metadataAuthority;
+    // A resource identifier to access the NFT. It could be any other data to be saved on the blockchain
+    string private uri;
+
+    // These events log on the blockchain transactions made with this NFT
+    event NFTMinted(address owner, address mintAccount);
+    event NFTSold(address from, address to);
+
+    // The mint account will identify the NFT in this example
+    constructor (address _mintAccount, address _metadataAuthority) {
+        mintAccount = _mintAccount;
+        metadataAuthority = _metadataAuthority;
+    }
+
+    /// Create a new NFT and associate it to an URI
+    ///
+    /// @param tokenURI a URI that leads to the NFT resource
+    /// @param mintAuthority an account that signs each new mint
+    /// @param ownerTokenAccount the owner associated token account
+    function createCollectible(string memory tokenURI, address mintAuthority, address ownerTokenAccount) public {
+        SplToken.TokenAccountData token_data = SplToken.get_token_account_data(ownerTokenAccount);
+
+        // The mint will only work if the associated token account points to the mint account in this contract
+        // This assert is not necessary. The transaction will fail if this does not hold true.
+        assert(mintAccount == token_data.mintAccount);
+        SplToken.MintAccountData mint_data = SplToken.get_mint_account_data(token_data.mintAccount);
+        // Ensure the supply is zero. Otherwise, this is not an NFT.
+        assert(mint_data.supply == 0);
+        
+        // An NFT on Solana is a SPL-Token with only one minted token.
+        // The token account saves the owner of the tokens minted with the mint account, the respective mint account and the number
+        // of tokens the owner account owns
+        SplToken.mint_to(token_data.mintAccount, ownerTokenAccount, mintAuthority, 1);
+        updateNftUri(tokenURI);
+        // Log on blockchain records information about the created token
+        emit NFTMinted(token_data.owner, token_data.mintAccount);
+    }
+
+    /// Transfer ownership of this NFT from one account to another
+    ///
+    /// @param oldOwnerTokenAccount the associated token account for the current owner
+    /// @param newOwnerTokenAccount the associated token account for the new owner
+    function transferOwnership(address oldOwnerTokenAccount, address newOwnerTokenAccount) public {
```

Comment:
nit: if you want to be very clear, you can rename these parameters since the owner may actually be the same
```suggestion
    function transferOwnership(address oldTokenAccount, address newTokenAccount) public {
```